components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -452,95 +461,141 @@ private String getOIDCAuthzEndpoint(Map<String, String> authenticatorProperties)
+     * Overridden in Google Authenticator for Google one tap.
+     *
+     * @param context       AuthenticationContext.
+     * @param request       HttpServletRequest
+     * @param oAuthResponse OAuthClientResponse
+     * @return The valid JWT token for the authentication request
+     * @throws AuthenticationFailedException when ID token is not valid. i.e Google Authenticator.
+     */
+    protected String mapIdToken(AuthenticationContext context, HttpServletRequest request,
+                                OAuthClientResponse oAuthResponse) throws AuthenticationFailedException {
 
-                if (log.isDebugEnabled() && IdentityUtil
-                        .isTokenLoggable(IdentityConstants.IdentityTokens.USER_ID_TOKEN)) {
-                    log.debug("Retrieved the User Information:" + jsonObject);
-                }
+        return oAuthResponse.getParam(OIDCAuthenticatorConstants.ID_TOKEN);
+    }
 
-                String authenticatedUserId = getAuthenticatedUserId(context, oAuthResponse, jsonObject);
-                String attributeSeparator = getMultiAttributeSeparator(context, authenticatedUserId);
+    /**
+     * Retrieves or maps the access token according to the flow supported by the authenticator.
+     * Overridden in Google Authenticator for Google one tap.
+     *
+     * @param request       HttpServletRequest.
+     * @param context       AuthenticationContext.
+     * @param oAuthResponse OAuthClientResponse.
+     * @throws AuthenticationFailedException Throws error when access token is not found.
+     */
+    protected void mapAccessToken(HttpServletRequest request, AuthenticationContext context,
+                                  OAuthClientResponse oAuthResponse) throws AuthenticationFailedException {
 
-                jsonObject.entrySet().stream()
-                        .filter(entry -> !ArrayUtils.contains(NON_USER_ATTRIBUTES, entry.getKey()))
-                        .forEach(entry -> buildClaimMappings(claims, entry, attributeSeparator));
-                
-                authenticatedUser = AuthenticatedUser
-                        .createFederateAuthenticatedUserFromSubjectIdentifier(authenticatedUserId);
-            } else {
+        String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
 
-                if (log.isDebugEnabled()) {
-                    log.debug("The IdToken is null");
-                }
-                authenticatedUser = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(
-                        getAuthenticateUser(context, jsonObject, oAuthResponse));
-            }
+        if (StringUtils.isBlank(accessToken)) {
+            log.error("No access token found");
+            throw new AuthenticationFailedException(ErrorMessages.ACCESS_TOKEN_EMPTY_OR_NULL.getCode(),
+                    ErrorMessages.ACCESS_TOKEN_EMPTY_OR_NULL.getMessage());
```

Comment:
Any specific reason why we are logging the error and throwing again?

Reply:
Fixed 9431b7b3c79135962b589fa4dd6934ccbfd2e2b0